smithy-openapi/src/main/java/software/amazon/smithy/openapi/OpenApiConfig.java

```diff
@@ -48,6 +45,34 @@ public enum HttpPrefixHeadersStrategy {
         WARN
     }
 
+    public enum OpenApiVersion {
+        /** OpenAPI versions supported by the converter. */
+        v3_0_2("3.0.2", true, true),
+        v3_1_0("3.1.0", false, false);
+
+        private final String version;
+        private final boolean supportsNullability;
+        private final boolean usesFormatKeywordForFilePayload;
+        OpenApiVersion(String version, boolean supportsNullability, boolean usesFormatKeywordForFilePayload) {
+            this.version = version;
+            this.supportsNullability = supportsNullability;
+            this.usesFormatKeywordForFilePayload = usesFormatKeywordForFilePayload;
+        }
+
+        @Override
+        public String toString() {
+            return version;
+        }
+
+        public boolean supportsNullability() {
+            return this.supportsNullability;
+        }
+
+        public boolean usesFormatKeywordForFilePayload() {
```

Comment:
Maybe this should be `supportsContentEncoding`

Reply:
Changed to `supportsContentEncodingKeyword` as per comment.